examples/solids/qfunctions/hyperSS.h

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2017, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-734707. All Rights
+// reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+/// @file
+/// Hyperelasticity, small strain for solid mechanics example using PETSc
+
+#ifndef HYPER_SS_H
+#define HYPER_SS_H
+
+#ifndef __CUDACC__
+#  include <math.h>
+#endif
+
+#ifndef PHYSICS_STRUCT
+#define PHYSICS_STRUCT
+typedef struct Physics_private *Physics;
+struct Physics_private {
+  PetscScalar   nu;      // Poisson's ratio
+  PetscScalar   E;       // Young's Modulus
+};
+#endif
+
+#ifndef LOG1P
+#define LOG1P
+// -----------------------------------------------------------------------------
+// Series approximation of log1p()
+//  log1p() is not vectorized in libc
+//
+//  The series expansion is accurate to 1e-7 in the range sqrt(2)/2 < J < sqrt(2),
+//  with machine precision accuracy near J=1.  The initialization extends this range
+//  to 0.35 ~= sqrt(2)/4 < J < sqrt(2)*2 ~= 2.83, which should be sufficient for
+//  applications of the Neo-Hookean model.
+// -----------------------------------------------------------------------------
+static inline CeedScalar log1p_series(CeedScalar x) {
+  const CeedScalar left = sqrt(2)/2 - 1, right = sqrt(2) - 1;
+  CeedScalar sum = 0;
+  if (1) { // Disable if the smaller range sqrt(2) < J < sqrt(2) is sufficient
+    if (x < left) { // Replace if with while for arbitrary range (may hurt vectorization)
+      sum -= log(2) / 2;
+      x = 1 + 2 * x;
+    } else if (right < x) {
+      sum += log(2) / 2;
+      x = (x - 1) / 2;
+    }
+  }
```

Comment:
Could flip this to `if (0)` because the SS formulation isn't valid outside that range anyway (due to geometric nonlinearity). Could also get away with fewer iterations, but that's premature optimization in my book.